dev/object_from_column_builder.h

```diff
@@ -44,5 +48,33 @@ namespace sqlite_orm {
                     })(column);
             }
         };
+
+        /**
+         *  Specialization for a table reference.
+         *  
+         *  This plays together with `column_result_of_t`, which returns `object_t<O>` as `table_referenece<O>`
+         */
+        template<class O, class DBOs>
+        struct struct_extractor<table_reference<O>, DBOs> {
+            const DBOs& db_objects;
+
+            O extract(const char* columnText) const = delete;
+
+            O extract(sqlite3_stmt* stmt, int&& /*nextColumnIndex*/) const {
```

Comment:
why `&&`?

Reply:
For the overload resolution, discerning it from the other overload taking `int&`.
It's because of a few things that play together.

There are two contexts in which row extractors are called:

1. as top-level extractor.
2. as embedded/nested extractor, i.e. as part of extracting other columns.

.
1. `storage_t::_execute_select()` is a generic implementation and always passes `0` to any top-level row extractor. It needs to do that because the value extractors always expect the column index, so we need a default argument for the structure extractors. So the argument is a temporary int [I just forgot the default value 0], which the structure extractors just discard, assuming they are only called from the top level.
2. `struct_extractor<tuple<...>>` in turn carries on the column index across multiple columns. In this case the other overload taking an `int&` is called.

I made a note on all the extraction methods.